### PR TITLE
With mock keycloak user

### PR DESCRIPTION
**Pull request by Kerruba**
Split the keycloak-connector into 2 modules and add testutils classes to enable mock Keycloak access tokens with a dedicated annotation similar to `WithMockUser`

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Review by ampie — CHANGES_REQUESTED** [review]
It was a good idea to assist plugin developers with testing infrastructure. The project structure makes sense too. My only request is for  the .idea directory to be removed from git and added to .gitignore.

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
-.idea/
+../.idea/
```
Comment (sergiofilhowz) [inline .gitignore]: I think this should be just .idea/

File: testutils/.gitignore [hunk 2]
```diff
@@ -0,0 +1,31 @@
+HELP.md
```
Comment (sergiofilhowz) [inline testutils/.gitignore]: we should merge this file with the one on root

**Review by sergiofilhowz — COMMENTED** [review]
README and LICENSE files should be on root folder

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Review by sergiofilhowz — APPROVED** [review]

**Comment by sergiofilhowz** [comment]
looks good

**Review by ampie — APPROVED** [review]

**Pull request merged**